src/Server/BaseServer.ts

```diff
@@ -246,46 +255,55 @@ export abstract class BaseServer implements IServer {
     this.programs.push(program);
   }
 
+  writeToScriptFile(path: ScriptFilePath, content: string) {
+    return this.#writeContentToFile(path, content, this.scripts, Script);
+  }
+
+  writeToTextFile(path: TextFilePath, content: string) {
+    return this.#writeContentToFile(path, content, this.textFiles, TextFile);
+  }
+
   /**
-   * Write to a script file
+   * Write to a Script or TextFile
    * Overwrites existing files. Creates new files if the script does not exist.
    */
-  writeToScriptFile(filename: ScriptFilePath, code: string): writeResult {
-    // Check if the script already exists, and overwrite it if it does
-    const script = this.scripts.get(filename);
-    if (script) {
-      // content setter handles module invalidation
-      script.content = code;
-      return { overwritten: true };
-    }
-
-    // Otherwise, create a new script
-    const newScript = new Script(filename, code, this.hostname);
-    this.scripts.set(filename, newScript);
-    return { overwritten: false };
+  writeToContentFile(path: ContentFilePath, content: string): writeResult {
+    const isTextFile = hasTextExtension(path);
+    if (isTextFile) return this.writeToTextFile(path, content);
+    return this.writeToScriptFile(path, content);
   }
 
-  // Write to a text file
-  // Overwrites existing files. Creates new files if the text file does not exist
-  writeToTextFile(textPath: TextFilePath, txt: string): writeResult {
-    // Check if the text file already exists, and overwrite if it does
-    const existingFile = this.textFiles.get(textPath);
-    // overWrite if already exists
+  #writeContentToFile<T extends ContentFilePath>(
```

Comment:
A couple things about this:
1. Rereading this, I agree that it's reached a point where it makes sense to have a helper method. Which I guess is why I suggested it originally. XD
2. It doesn't actually need to be private. The only member you use off `this` that I see is `hostname`, I think it'll work out better to just pass that as well and then have it be a plain-old (genericized) function, as opposed to any kind of method.